aws/internal/keyvaluetags/sso_tags.go

```diff
@@ -0,0 +1,95 @@
+// +build !generate
+
+package keyvaluetags
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssoadmin"
+)
+
+// Custom SSO tag service functions using the same format as generated code.
```

Comment:
let's use the generator available in the provider to get these functions, of course with the exception that we need to account for the `InstanceArn` param .. following along here should be a good start: https://github.com/hashicorp/terraform-provider-aws/blob/master/docs/contributing/contribution-checklists.md#adding-service-to-tag-generating-code

to get the `InstanceArn` param in place, going to the `service_generation_customizations.go` file and then the the func `ServiceTagResourceTypeField(serviceName string)` , there you can add the customization like:
```go
func ServiceTagResourceTypeField(serviceName string) string {
	switch serviceName {
	...
	case "ssm":
		return "ResourceType"
	case "ssoadmin":
		return "InstanceArn"
	default:
		return ""
	}
}
```